### PR TITLE
lightbox

### LINKED ISSUE #7469: [Core appliances pages] Implement lightbox
Copy the lightbox code used on snapstore snap pages

### PR DESCRIPTION
**Pull request by deadlight**
## Done

- Added lightbox JS
- Added lightbox to Plex

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/appliance/plex
- Click on the images and test the lightbox
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)

## Issue / Card

Fixes #7469

### CODE REVIEW

File: templates/appliance/plex/index.html [hunk 1]
```diff
@@ -63,9 +71,9 @@ <h1 class="p-heading--three u-no-margin--bottom">
   <div class="row">
     <div class="col-8">
       <h2 class="p-heading--four">Plex magically organizes your media libraries and streams them to any device</h2>
-      <p>Plex magically organizes your media libraries and streams them to any device &mdash; including all your video, music, and photo libraries. With a Plex Pass, supported tuner, and digital antenna, you can also watch and record FREE over-the-air broadcast TV, including major networks. You can also enjoy podcasts Plus Plex News &mdash; the most balanced and comprehensive personalized video newsfeed out there. </p>
```
Comment (anthonydillon) [inline templates/appliance/plex/index.html:66]: Looks like you reverted some stuff here

File: webpack.config.js [hunk 2]
```diff
@@ -22,6 +22,8 @@ module.exports = {
     "tco-calculator": "./static/js/src/tco-calculator.js",
     "sticky-nav": "./static/js/src/sticky-nav.js",
     imageBuilder: "./static/js/src/imageBuilder.js",
+    lightbox: "./static/js/src/lightbox.js",
+    appliance: "./static/js/src/appliance.js"
```
Comment (anthonydillon) [inline webpack.config.js:26]: The JS is not added to the page at all.

Comment (deadlight) [reply webpack.config.js:26]: ugh, I somehow blatted my changes with the rebase. Hold on

**Comment by deadlight** [comment]
@anthonydillon fixed the rebase issues.

File: static/sass/styles.scss [hunk 3]
```diff
@@ -67,6 +67,7 @@ $color-shadow: rgba(0, 0, 0, 0.5) !default;
 @import "utility-animations";
 @import "utility_crop";
 @import "utility_vertically-center";
+@import "pattern_lightbox";
```
Comment (anthonydillon) [inline static/sass/styles.scss:70]: Can you make this alphabetical please

File: templates/appliance/plex/index.html [hunk 4]
```diff
@@ -162,4 +170,6 @@ <h2>Build your own appliance</h2>
   </div>
 </section>
 
-{% endblock content %}
+<script src="{{ versioned_static('js/build/appliance.min.js') }}" defer></script>
```
Comment (anthonydillon) [inline templates/appliance/plex/index.html:173]: Can you add this via the appliance base template so this script is used across the pages.

File: webpack.config.js [hunk 5]
```diff
@@ -22,6 +22,8 @@ module.exports = {
     "tco-calculator": "./static/js/src/tco-calculator.js",
     "sticky-nav": "./static/js/src/sticky-nav.js",
     imageBuilder: "./static/js/src/imageBuilder.js",
+    lightbox: "./static/js/src/lightbox.js",
```
Comment (anthonydillon) [inline webpack.config.js:25]: Why do you need to build a standalone version of `lightbox` when its included in applance.js?

**Comment by spencerbygraves** [comment]
@deadlight when I click the top right image, the light box loads but that first image is shown twice.

I see it, click the arrow, see it again.

image

**Comment by deadlight** [comment]
ok, that was a weird one to debug. It's caused by GA rewriting the hrefs so that the JS can't match them properly. I think I can catch and sanitise the parameters.

**Comment by deadlight** [comment]
@anthonydillon I had to add a couple of lines of JS to fix the bug. Can you check that it's still good, please?

@spencerbygraves good catch on the bug, I've fixed it

**Comment by anthonydillon** [comment]
## QA
I'm getting an issue with the image loading when clicking on one. Its blank and you need to hit the next button to show the image.

**Comment by deadlight** [comment]
@anthonydillon is there a specific image that's triggering the bug? I'm struggling to replicate the issue. There might be some edge case I'm not seeing.

**Comment by anthonydillon** [comment]
@deadlight after checking again it worked well sorry about that.

**Pull request merged**